src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -737,9 +749,6 @@ private async Task ProcessRequests<TContext>(IHttpApplication<TContext> applicat
                 {
                     await messageBody.StopAsync();
                 }
-
-                // Clear any AsyncLocals set during the request; back to a clean state ready for next request
-                ExecutionContext.Restore(cleanContext);
```

Comment:
The execution context capture at the start of ProcessRequest is no longer restored here.

Is that ok because when ProcessRequest is called again the previously captured context with `InitialExecutionContext = ExecutionContext.Capture();` will be restored?

Reply:
Right. We now restore at the beginning of the of the ProcessRequests loop instead of the end.

If the loop continues as usually is usually the case with HTTP/1.1, the restore happens at effectively the same time as there are no `awaits` in between.

In the case of HTTP/2, the restore now happens right after the potentially-pooled Http2Stream gets dispatched to the ThreadPool. Restoring the ExecutionContext immediately before ProcessRequests exiting served no purpose since the AsyncStateMachine gets GC'd immediately after anyway.